build/Dockerfile.k8s-cloud-builder

```diff
@@ -47,7 +47,5 @@ RUN \
       stable edge" && \
    apt-get -y update
 
-# As of 2018-04-10, leaving this pinned to 17.09 due to image pull issues
-# with 17.12
-ARG DOCKER_VERSION=17.09.0~ce-0~ubuntu
+ARG DOCKER_VERSION=18.06.0~ce~3-0~ubuntu
```

Comment:
`18.06.1~ce~3-0~ubuntu` is already out, just see if you can use that version..

Reply:
@mkumatag we kinda tested the 18.06.0 bump yesterday so let's leave it alone unless 18.06.1 is absolutely necessary

Reply:
this same pinning is in place in test-infra for docker in docker building (we talked to releng about this) because we started flaking a ton on image pull around then. we've also not upgraded.. filing a bug for us to upgrade as well